cli/tools/registry/mod.rs

```diff
@@ -158,7 +161,7 @@ async fn prepare_publish(
     collect_excluded_module_diagnostics(
       &root_specifier,
       &graph,
-      file_patterns.as_ref(),
+      &file_patterns,
```

Comment:
Instead of doing this, it should instead collect all the files for the tarball, check if they're excluded, then pass the paths to create_gzipped_tarball.